<robot name="homework_robot">
  
  <material name="red">
    <color rgba="1 0 0 0.8"/>
  </material>
  
  <material name="yellow">
    <color rgba="1 1 0 0.8"/>
  </material>

## links
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="1.5" radius="0.4"/>
      </geometry>
      <material name="red"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="1.5" radius="0.4"/>
      </geometry>
    </collision>
  </link>

  <link name="body">
    <visual>
      <geometry>
        <box size="0.5 0.5 0.5"/>
      </geometry>
      <origin xyz="0 0 0.25"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.5 0.5 0.5"/>
      </geometry>
    </collision>
  </link>

  <link name="arm">
    <visual>
      <geometry>
        <cylinder length="0.7" radius="0.2"/>
      </geometry>
      <origin xyz="0 0 0.35"/>
      <material name="red"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.7" radius="0.2"/>
      </geometry>
    </collision>
  </link>

  <link name="end">
    <visual>
      <geometry>
        <sphere radius="0.3"/>
      </geometry>
      <origin xyz="0 0 0.25"/>
      <material name="yellow"/>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.3"/>
      </geometry>
    </collision>
  </link>

## joints

  <joint name="base_to_body" type="prismatic">
    <parent link="base_link"/>
    <child link="body"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-0.2" upper="1.5" velocity="0.3"/>
    <origin xyz="0 0 0.75"/>
  </joint>

  <joint name="body_to_arm" type="revolute">
    <parent link="body"/>
    <child link="arm"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5"/>
    <origin xyz="0 0 0.6"/>
  </joint>

  <joint name="arm_to_end" type="prismatic">
    <parent link="arm"/>
    <child link="end"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="2.0" velocity="0.4"/>
    <origin xyz="0 0 0.65"/>
  </joint>

</robot>

